internal/hcpvalidator/resource_type.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package hcpvalidator
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/helpers/validatordiag"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+const (
+	validResourceTypeRegex = `^hashicorp(\.([a-z][a-z-]*)){2}$`
+	invalidResourceTypeErr = "a Resource Type must have format hashicorp.<service>.<resource> and consist only of lowercase alphabetic characters, dashes, and dots"
+)
+
+var (
+	resourceTypeRegex                  = regexp.MustCompile(validResourceTypeRegex)
+	_                 validator.String = resourceTypeValidator{}
+)
```

Comment:
Looking at the validators in this PR I'm guessing this is convention used in the provider for checking that the type implements the ValidateString interface, correct? 

I found it a bit odd seeing as the type is actually being defined below. It feels like a check that could be added as as simple unit test to check if it implements the interface, especially since the compiler will push back with a type check error if  `resourceTypeValidator.ValidateString` is being passed/called and it no longer implements the interface. 

If this must be within the validator code for convention my nit would be to use two separate var definitions to remove the added whitespace from fmt. 



```suggestion
var _ validator.String = resourceTypeValidator{}
var resourceTypeRegex = regexp.MustCompile(validResourceTypeRegex)
```